File: leetcode/golang/53.最大子序和.go
func maxSubArray(nums []int) int {
    if len(nums) == 0 {
        return 0;
    };

    ret := nums[0];
    for i, _ := range nums {
        if i == 0 {
            continue;
        }
        if nums[i] + nums[i - 1] > nums[i] {
            nums[i] = nums[i] + nums[i - 1];
        }
        if nums[i] > ret {
            ret = nums[i];
        }
    }
    return ret;
}

File: leetcode/python/53.最大子序和.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        ret = nums[0]
        for i in range(1, len(nums)):
            nums[i] = max(nums[i] + nums[i - 1], nums[i])
            ret = max(ret, nums[i])
        return ret